urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/http_server/HttpServer.java

```diff
@@ -0,0 +1,73 @@
+package terminal.http_server;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Palachsuper
+ * Date: 26.05.14
+ * Time: 11:15
+ * To change this template use File | Settings | File Templates.
+ */
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import terminal.devices.cashe_machine;
+import terminal.service.Service;
+/**
+ * * Обрабатывает запросы от клиентов, возвращая файлы, указанные
+ * * в url-path или ответ с кодом 404, если такой файл не найден. *
+ * */
+public class HttpServer {
+    /** * Первым аргументом может идти номер порта. */
+    public static Thread t;
+    public static boolean a;
+    public static void main(String args) {
+    /* Если аргументы отсутствуют, порт принимает значение поумолчанию */
+        int port = 20143;
+        /* Создаем серверный сокет на полученном порту */
+        ServerSocket serverSocket = null;
+        try { serverSocket = new ServerSocket(port);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *serverSocket*. Execution paths that do not contain closure statements were detected. Ensure *serverSocket* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).